### PR TITLE
feat: disabled eggs being thrown by dispensers

### PR DESCRIPTION
**Pull request by rooooose-b**

### CODE REVIEW

**Comment by rooooose-b** [comment]
Using `dropItem` to spawn in the egg was the best I could find, though it looks a bit strange when they spawn in. `dropItemNaturally` could potentially be better, but then they could be thrown towards the dispenser which would also look weird. There's probably a better solution that I just didn't think of when making this.

File: src/main/java/me/ShermansWorld/AlathraExtras/disabledispensereggs/DispenserListener.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockDispenseEvent;
+import org.bukkit.inventory.ItemStack;
+
+public class DispenserListener implements Listener {
+
+    @EventHandler
+    public void onDispense(BlockDispenseEvent e) {
+        if (e.getBlock().getType() == Material.DISPENSER) {
```
Comment (darksaid98) [inline src/main/java/me/ShermansWorld/AlathraExtras/disabledispensereggs/DispenserListener.java:15]:
```suggestion
        if (e.getBlock().getType().equals(Material.DISPENSER)) {
```

File: src/main/java/me/ShermansWorld/AlathraExtras/disabledispensereggs/DispenserListener.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            if (e.getItem().getType() == Material.EGG) {
```
Comment (darksaid98) [inline src/main/java/me/ShermansWorld/AlathraExtras/disabledispensereggs/DispenserListener.java:16]:
```suggestion
            if (e.getItem().getType().equals(Material.EGG)) {
```

**Review by darksaid98 — CHANGES_REQUESTED** [review]
Not a fan of how the item is dropped. Would it be possible to copy the velocity and position of the egg projectile and apply it to the item? It would shoot them really far but at least they wouldn't hatch. 

You might also be able to delete the projectile without canceling the entire event (*Which would then allow you to rely on the original item consumption mechanic*).

File: src/main/java/me/ShermansWorld/AlathraExtras/disabledispensereggs/DispenserListener.java [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
 public class DispenserListener implements Listener {
 
     @EventHandler
-    public void onDispense(BlockDispenseEvent e) {
-        if (e.getBlock().getType() == Material.DISPENSER) {
-            if (e.getItem().getType() == Material.EGG) {
+    public void onProjectileLaunch(ProjectileLaunchEvent e) {
+        if (e.getEntity().getShooter() instanceof BlockProjectileSource) {
+            if (e.getEntity().getType().equals(EntityType.EGG)) {
+                Bukkit.getLogger().info("Egg detected.");
```
Comment (darksaid98) [inline src/main/java/me/ShermansWorld/AlathraExtras/disabledispensereggs/DispenserListener.java:21]:
```suggestion
```
Let's leave the debug log out :)

**Review by darksaid98 — CHANGES_REQUESTED** [review]
Other than that, looks good. Exceptional work with the PR's.

**Review by darksaid98 — APPROVED** [review]

**Pull request merged**